src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceMultinomialLogit.scala

```diff
@@ -46,24 +49,44 @@ class ModeChoiceMultinomialLogit(
     destinationActivity: Option[Activity],
     person: Option[Person] = None
   ): Option[EmbodiedBeamTrip] = {
+
+    def findBestIn(group: IndexedSeq[ModeCostTimeTransfer]): ModeCostTimeTransfer = {
+      if (group.size == 1) {
+        group.head
+      } else if (group.head.embodiedBeamTrip.tripClassifier.isTransit) {
+        val inputData = group
+          .map(
+            mct => mct.embodiedBeamTrip -> attributes(timeAndCost(mct), mct.transitOccupancyLevel, mct.numTransfers)
+          )
+          .toMap
+        val alternativesWithUtility = model.calcAlternativesWithUtility(inputData)
+        val chosenModeOpt = model.sampleAlternative(alternativesWithUtility, random)
+        chosenModeOpt
+          .flatMap(sample => group.find(_.embodiedBeamTrip == sample.alternativeType))
+          .getOrElse(group minBy timeAndCost)
+      } else {
+        group minBy timeAndCost
+      }
+    }
+
     if (alternatives.isEmpty) {
       None
     } else {
       val modeCostTimeTransfers = altsToModeCostTimeTransfers(alternatives, attributesOfIndividual, destinationActivity)
 
       val bestInGroup =
-      modeCostTimeTransfers groupBy (_.mode) map {
-        case (_, group) => group minBy timeAndCost
+      modeCostTimeTransfers groupBy (_.embodiedBeamTrip.tripClassifier) map {
+        case (_, group) => findBestIn(group)
       }
       val inputData = bestInGroup.map { mct =>
         val theParams: Map[String, Double] =
-          Map("cost" -> (mct.cost + mct.scaledTime))
-        val transferParam: Map[String, Double] = if (mct.mode.isTransit) {
-          Map("transfer" -> mct.numTransfers)
+          Map("cost" -> timeAndCost(mct))
+        val transferParam: Map[String, Double] = if (mct.embodiedBeamTrip.tripClassifier.isTransit) {
+          Map("transfer" -> mct.numTransfers, "transitOccupancyLevel" -> mct.transitOccupancyLevel)
```

Comment:
And here rather than doing the utility calculations again we can just use `expectedMaximumUtility` from above as the utility of any options for which the best alternative was already chosen, e.g. as a new parameter "utility" with multiplier = 1

Reply:
Created issue #2863